packages/cli/src/cmds/agentInstaller/projectConfiguration.ts

```diff
@@ -81,12 +84,42 @@ async function resolveSelectedInstallers(
     }
   }
 
-  // Sanity check. Verify everything resolved properly. This should branch should never occur.
+  // Sanity check. Verify everything resolved properly. This branch should never occur.
   if (projects.some(({ selectedInstaller }) => !selectedInstaller)) {
     throw new Error('Invalid selection');
   }
 }
 
+export function getYarnPackages(availableInstallers: AgentInstaller[]): string[] {
+  let yarnPackages: string[] = [];
+  availableInstallers.forEach((installer) => {
+    if (installer.name === 'yarn') {
+      const packageJsonFilename = join(installer.path, 'package.json');
+      if (existsSync(packageJsonFilename)) {
+        const data = readFileSync(packageJsonFilename, 'utf-8');
```

Comment:
I'm nit-picking here, but we should be opting for the asynchronous versions of these methods. It'll allow other work to be processed while it hits the disk.

Reply:
I went ahead and changed this.  Initially I tried `const data = await fs.readFile(packageJsonFilename, 'utf-8');` but the resulting `data` couldn't get processed by `JSON.parse` until I switched the loop from `.forEach` to `for (const ...) {`.

But then I used synchronous `lstatSync` because async `lstat` didn't seem to return `isDirectory()`.